sprockets/mixins/avro_publisher/mixins.py

```diff
@@ -0,0 +1,180 @@
+import io
+import json
+import logging
+
+from sprockets.mixins import amqp
+from tornado import gen, httpclient
+import fastavro
+
+LOGGER = logging.getLogger(__name__)
+
+
+class PublishingMixin(amqp.PublishingMixin):
+    """The request handler will connect to RabbitMQ on the first request,
+    blocking until the connection and channel are established. If RabbitMQ
+    closes its connection to the app at any point, a connection attempt will
+    be made on the next request.
+
+    This class implements a pattern for the use of a single AMQP connection
+    to RabbitMQ.
+    """
+    DATUM_MIME_TYPE = 'application/vnd.apache.avro.datum'
+    DEFAULT_SCHEMA_URI_FORMAT = 'http://localhost/avro/%(name)s.avsc'
+    DEFAULT_FETCH_RETRY_DELAY = 0.5
+
+    def initialize(self, *args, **kwargs):
+        self._schema_fetch_failed = False
+        self._http_client = httpclient.AsyncHTTPClient(force_instance=True)
```

Comment:
It's probably a good idea to see if self._http_client already exists before setting it.